main.go

```diff
@@ -78,6 +76,31 @@ func main() {
 		regArr = append(regArr, re)
 	}
 
+	if len(flag.Args()) == 0 {
+		getBuf(os.Stdin)
+	} else {
+		for _, j := range flag.Args() {
```

Comment:
What is j?

Reply:
depends on the command i run but j in the first iteration is flag.Args()[0] and so on

Reply:
"depends on the command"? On which command?

Reply:
it depends on the files i specify in go run . .... toGrep.txt ...

Reply:
"files". In short, j is a file name. Name the variable accordingly.